Chat_3/Program.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Chat_3
+{
+    class Program
+    {
+        private static string serverIp = "127.0.0.1";
+        private static Socket mainSocket;
+        private static bool isConnecting = false;
+        private static bool isDelivered = false;
+        private static uint counter = 0;
+
+        static void Main(string[] args)
+        {
+            Console.Write("Enter the port to receive messages: ");
+            int localPort = Int32.Parse(Console.ReadLine());
+            Console.Write("Enter the port for sending messages: ");
+            int remotePort = Int32.Parse(Console.ReadLine());
+            Console.WriteLine("Waiting for connection");
+            mainSocket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
+            try
+            {
+                EndPoint remotePoint = new IPEndPoint(IPAddress.Parse(serverIp), remotePort);
+                IPEndPoint localIP = new IPEndPoint(IPAddress.Parse(serverIp), localPort);
+                mainSocket.Bind(localIP);
+                while (!isConnecting)
+                {
+                    Connect(remotePoint); 
+                    Thread.Sleep(300);
+                }
+
+                Task listeningTask = new Task(Listen);
+                listeningTask.Start();
+
+                while (true)
+                {
+                    Send(Console.ReadLine(), remotePoint);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Main: " + ex.Message);
+            }
+        }
+
+        private static void Send(string message, EndPoint remotePoint)
+        {
+            counter++;
+            isDelivered = true;
+            var data = Encoding.Unicode.GetBytes(message + "|" + counter);
+            while (isDelivered)
+            {
+                mainSocket.SendTo(data, remotePoint);
+                Thread.Sleep(300);
+            }

```

Comment:
Can you make it asynchronous? So that user is able to send messages even before the delivery status is changed?